cupsfilters/ipp.c

```diff
@@ -262,10 +262,10 @@ get_printer_attributes5(http_t *http_printer,
   get_printer_attributes_log[0] = '\0';
 
   /* Convert DNS-SD-service-name-based URIs to host-name-based URIs */
-  if(!isFax)
+  if(is_fax == -1)
     uri = resolve_uri(raw_uri);
```

Comment:
Why do you use resolve_uri() for print URIs and ippfind_based_uri_converter() for fax URIs? You could use ippfind_based_uri_converter() for both. Or is resolve_uri() faster due to the fact that it does not call an external executable?
Another possibility would be to make the caller somehow select which of the two functions is used.
Currently cups-browsed could use resolve_uri()  and driverless could use ippfind_based_uri_converter(), but later you will also add fax support to cups-browsed.
If there is actually a performance penalty by calling the external `ippfind` executable then it would be best to do all print URI conversions with resolve_uri() for performance and all fax URI conversions with ippfind_based_uri_converter() for fax support.

Reply:
>Why do you use resolve_uri() for print URIs and ippfind_based_uri_converter() for fax URIs? You could use ippfind_based_uri_converter() for both. Or is resolve_uri() faster due to the fact that it does not call an external executable?

In `utils/driverless.c` file ,`isFax` is set to "1" if we run commands with `driverless-fax` else `isFax` is set to "0"(for only `driverless` calls).So ,`ippfind_based_uri_converter()` is for both - print uris and as well fax uris. `is_fax` is set to "-1" (while calling `get_printer_attributes5()` from `get_printer_attributes3()`) for calls from `cups-browsed`, so that `cups-browsed` use `resolve_uri`. 
>If there is actually a performance penalty by calling the external ippfind executable then it would be best to do all print URI conversions with resolve_uri() for performance and all fax URI conversions with ippfind_based_uri_converter() for fax support.
 
I have used [clock()](http://www.gnu.org/software/libc/manual/html_node/CPU-Time.html) function available in time.h (or is there any better way to check?)to performance check . But both `ippfind_based_uri_converter()` and `resolve_uri()` were giving approximately the same time (~0.026 sec).